### ISSUE TITLE
Creating a /v3/deployment causes /v2/spaces/:guid/summary to show the wrong app guid

### ISSUE THREAD

**Issue by forbushbl** [cloudfoundry/cloud_controller_ng#1523]
## Issue

When you create a rolling deployment for an app using `/v3/deployments`, the `/v2/spaces/:guid/summary` is incorrectly updated with the process guid in the `guid:` field instead of the app guid.

## Context

We are running capi-release 1.87.0

## Steps to Reproduce

1. Create an app in Cloud Foundry
2. Check the app's guid (`cf app --guid`) and check the space summary ('cf curl /v2/spaces/:guid/summary`) and note that the GUID for the app matches
3. Create a deployment for the app. 
   ```
   cf curl -X POST /v3/deployments -d '{"droplet":{"guid":"5ef19cb0-555d-43c0-8e13-226ad8a749fa"},"strategy":"rolling","relationships":{"app":{"data":{"guid":"2f08866a-81bc-44b0-b542-8621c9fb3f58"}}}}'
   ```
4. Check the app's guid (`cf app --guid`) and check the space summary ('cf curl /v2/spaces/:guid/summary`) and note that the GUID for the app **<ins>no longer matches</ins>**. Further investigation reveals that the app guid in the v2 space summary is actually the process guid for the app.

## Expected result

I would expect that updating an app using the v3 API would still result in the v2 space summary returning the correct app guid.

## Current result

As noted in the steps to reproduce, the v2 space summary incorrectly shows the process_guid in the app's guid field.

**Comment by cwlbraa** [comment]
This is a known behavior, actually, and not precisely a bug.

As I'm sure you know, one of the biggest changes from v2 to v3 is the introduction of `v3/Process`es within a `v3/App`. One of the big goals behind the gradual v3 migration process was to maintain interoperability between the 2 api versions. As a consequence of that goal, the `v2/apps` table actually became the `v3/processes` table and a different `v3/apps` table was added to contain those processes. 

When you look at the `v2/apps` guid, what you're actually seeing is a `v3/process` guid, and when apps are initially created the web process' guid matches its `v3/app` guid. When you perform a deployment, a second process with a new guid is introduced in the process of rolling out your new code and configuration, at which point the `v3/processes` guid no longer matches the `v2/app` guid. `cf app --guid NAME` is giving you a v3 app guid, but the space summary gives a v2 one. 

We recommend using exclusively v3 endpoints to avoid this sort of weirdness, as v2 will be going away soon. If you're using the cli, the cf7 beta also avoids these sorts of issues.

We don't plan on fixing this before deprecating v2.

**Comment by voelzmo** [comment]
Thanks @cwlbraa for the explanation!

I'd like to take up the conversation regarding one point specifically
> We recommend using exclusively v3 endpoints to avoid this sort of weirdness, as v2 will be going away. If you're using the cli, the cf7 beta also avoids these sorts of issues.

How would you imagine this to work in a world where a platform provider has a UI (e.g. currently working with v2 API, especially due to some performance shortcomings in v3) and consumers of the platform which can independently decide whether they're using `cf` or `cf7`?

**Comment by voelzmo** [comment]
Some additional context, so I don't have to dig this up again for future discusisons:

The issue comes up, once you have re-deployed an app with `v3`, due to the reasons pointed out by @cwlbraa above.

* `/v2/spaces/<GUID>/summary` returns `v3/process` GUIDs as `app.guid`
* `/v2/spaces/<GUID>/apps` returns `v3/app` GUIDs as `app.guid`
* `/v2/spaces/<GUID>/apps` returns url references with `v3/process` GUIDs in their URL, e.g. `"routes_url": "/v2/apps/c21f34fd-b9e3-462a-9658-f121005c57bb/routes",`

This seems inconsistent at best, even if you'd like to technically not label this as a bug, as stated above. I guess, v2 API should either
* point to `v3/process` GUIDs *or*
* point to `v3/app` GUIDs 
*but* most likely not mix this – especially not in a single call like `/v2/spaces/<GUID>/apps`

I understand the current focus on finishing v3, at the same time I'd like to understand more about the expectations we can have regarding compatibility between v2/v3.

/cc @ssisil @emalm @Gerg

**Comment by cwlbraa** [comment]
💯 I didn't know that there were v2 endpoints returning actual app guids. 

You also make a good point about the goals of the interoperability. I'm gonna label this a bug because I agree it should be consistent. 

I guess what I was trying to get across was that deployments are one of the first APIs where we know we're diverging pretty aggressively from existing v2 semantics and so it's somewhat expected that some rough edges will appear.

**Comment by voelzmo** [comment]
Thanks @cwlbraa for re-opening this and sharing your assessment!

**Comment by Gerg** [comment]
I believe all the cases from this issue have been addressed in #1620 and #1716. The changes should be included in [capi-release 1.97](https://github.com/cloudfoundry/capi-release/releases/tag/1.97.0). I'm going to close this issue, but feel free to re-open it if you are still having issues.
